### ISSUE TITLE
housekeeping to prep for infrastructure authoring

### ISSUE THREAD

**Issue by civilx64** [IfcOpenShell/IfcOpenShell#5602]
With infrastructure geometry processing now in place in the C++ core layer, I need to do some cleanup that is still hanging around from the initial python prototypes.

**Comment by RickBrice** [comment]
I’ve been staring to pencil out an API for alignment. Will post soon. 

I could use a quick start tutorial for binding the C++ code with Python.

**Comment by aothms** [comment]
> I could use a quick start tutorial for binding the C++ code with Python.

Unfortunately the code in the IfOpSh repository is at most an example on how not to do things.

Maybe it's easiest if we do this collaboratively, start from the C++ and fill in the binding as we go.

We also have this https://github.com/buildingSMART/ifc-gherkin-rules/tree/main/features/steps/utils/ifc43x_alignment_validation

**Comment by civilx64** [comment]
I did some initial investigation into [pybind11](https://pybind11.readthedocs.io/en/stable/).  I was able to get the basic example to compile but did not make much progress after that.

I would recommend that we move forward with alignment authoring in a similar way to what was successful with the import and geometry generation: 

1. prototype and debug in python
2. port to C++ and create python bindings

There will likely be some parallel efforts and overlaps as this plays out.  For example, in the short term it's quicker IMO to do a python port of IfcAlignmentHelper versus working through the config for binding an extension module.
